src/rpdk/core/contract/type_configuration.py

```diff
@@ -6,48 +6,50 @@
 
 LOG = logging.getLogger(__name__)
 
-TYPE_CONFIGURATION_FILE_PATH = "~/.cfn-cli/typeConfiguration.json"
-
 
 class TypeConfiguration:
     TYPE_CONFIGURATION = None
 
     @staticmethod
-    def get_type_configuration():
+    def get_type_configuration(typeconfigloc):
+        if typeconfigloc:
+            type_config_file_path = typeconfigloc
+        else:
+            type_config_file_path = "~/.cfn-cli/typeConfiguration.json"
```

Comment:
I think `~` fails in some scenarios. We should use the standard function for getting the user's home directory on all platforms.

Reply:
this is expanded by python's os.path.expanduser function and works across posix and windows.